lib/handlers/events/delivery/goals/GoalAutomationEventListener.ts

```diff
@@ -93,14 +93,13 @@ export class GoalAutomationEventListener extends AutomationEventListenerSupport
                         }, reject);
                     });
                 });
-                safeExit(0);
             } catch (e) {
                 logger.error(`Processing goal failed: ${e.message}`);
-                // We always have to exit with 0 here as otherwise k8s will launch a new instance.
-                // FulfillGoalOnRequested throws an error in case of a failing goal.
-                // ^ TODO CD look at this and see if that makes any sense or can be changed
-                safeExit(0);
             }
+            // Exit successfully to avoid job handlers retrying.
+            // FulfillGoalOnRequested throws an error in case of a failing goal.
+            // ^ TODO CD look at this and see if that makes any sense or can be changed
```

Comment:
```suggestion
```

I think we can kill that comment now, no?

Reply:
It sounded like you thought FulfillGoalOnRequested throwing the error was the right thing, but wasn't sure so I left it in for your review. I'll remove it.

Reply:
I'm going to remove the comment about FulfillGoalOnRequested throwing an error too since that error never makes it this far up the stack.